tinygrad/renderer/assembly.py

```diff
@@ -84,6 +84,9 @@ def ptr_ar(root):
 
   matcher = PatternMatcher([
     ({"__name__": "root", "uop": UOps.ALU, "arg": BinaryOps.CMPEQ, "vin": ({"__name__": "x", "dtype": dtypes.bool},{"__name__": "y"})}, eq_rep),
+    ({"__name__": "root", "uop": UOps.ALU, "arg": BinaryOps.ADD, "dtype": set([dtypes.float16, dtypes.bfloat16, dtypes.float32, dtypes.float64]),
+      "vin": [{"__name__": "non_muls"}, {"__name__": "muls", "uop": UOps.ALU, "arg": BinaryOps.MUL}]},
+      lambda root, muls, non_muls : uops.add(UOps.ALU, root.dtype, muls.vin + (non_muls,), TernaryOps.MULACC, insert_before=uops.uops.index(root))),
```

Comment:
Do you need a lambda here? Can this just be the UOp itself?

Reply:
> Do you need a lambda here? Can this just be the UOp itself?

Do you mean changing this into the function?
```py
  def mulacc(root, muls, non_muls):
    root.arg = TernaryOps.MULACC
    root.vin = muls.vin + (non_muls,)
```

Reply:
No I mean returning the UOp instead of uops.add. Sorry, it's still a lambda.

Example:
```
  ({"uop": UOps.ALU, "arg": BinaryOps.ADD, "vin": ({"__name__": "x"}, {"__name__": "my", "uop": UOps.ALU, "arg": UnaryOps.NEG})},
    lambda x, my: UOp(UOps.ALU, x.dtype, (x, my.vin[0]), BinaryOps.SUB)),
```

Reply:
Oh that wouldn't work, replacing in PTX assumes that it's in `uops.uops`

```py
  for u in uops:
    if u in seen: continue
    seen.add(u)
    for o,n in replace.items():
      if o in u.vin and u is not n:
        u.vin = tuple(n if x == o else x for x in u.vin)
    if rew := matcher.rewrite(u): replace[u] = rew
```

Reply:
> Example:
> 
> ```
>   ({"uop": UOps.ALU, "arg": BinaryOps.ADD, "vin": ({"__name__": "x"}, {"__name__": "my", "uop": UOps.ALU, "arg": UnaryOps.NEG})},
>     lambda x, my: UOp(UOps.ALU, x.dtype, (x, my.vin[0]), BinaryOps.SUB)),
> ```

With the new pattern matcher rewrites this is now possible - updated